### PR TITLE
Update MLflow UI

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Update MLflow UI.

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests (describe details, including test results, below)

https://user-images.githubusercontent.com/17039389/207022893-2436d7d4-d2c3-47bc-8d33-9282baabefe8.mov

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/js/src/common/utils/FeatureUtils.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/**
+ * This file aggregates utility functions for enabling features configured by flags.
+ * In the OSS version, you can override them in local development by manually changing the return values.
+ */
+/**
+ * UI feature preview: new ("next") version of runs comparison view within experiment runs table.
+ */
+export const shouldUseNextRunsComparisonUI = () => false;
```
Comment (hubertzub-db) [inline mlflow/server/js/src/common/utils/FeatureUtils.ts:8]: So the next-gen run comparison feature is still WIP, but there are already a few half-made views there - and it's possible to flip a "flag" right here in order to see them in OSS mlflow as well. Is it reasonable to retain it here so the OSS community will get some insight as well? Or shall we remove it from this update? WDYT @dbczumar @harupy

Comment (dbczumar) [reply mlflow/server/js/src/common/utils/FeatureUtils.ts:8]: @hubertzub-db I think it's fine to keep the flag here for OSS, as long as it's `false` for now :)

File: mlflow/server/js/package.json [hunk 2]
```diff
@@ -116,16 +121,18 @@
...
     "webpack": "^5.69.0"
   },
   "scripts": {
     "//": "IMPORTANT: For compatibility with Windows in MLflow OSS, do NOT inline environment variables in scripts other than `start-web`, `start:mfe` and `test:ci`. Environment variables can instead be set in `mlflow/server/js/scripts/global-setup.js`.",
     "start": "craco start",
     "start:iframe": "HIDE_HEADER=true HIDE_EXPERIMENT_LIST=true SHOW_GDPR_PURGING_MESSAGES=true USE_ABSOLUTE_AJAX_URLS=true SHOULD_REDIRECT_IFRAME=true craco start",
-    "start:dev-proxy": "MLFLOW_DEV_PROXY_MODE=true yarn start:iframe",
+    "start:dev-proxy-fs": "START_FEATURE_STORE=true yarn start:dev-proxy-iframe",
+    "start:dev-proxy-iframe": "MLFLOW_DEV_PROXY_MODE=true yarn start:iframe",
+    "start:dev-proxy": "MLFLOW_DEV_PROXY_MODE=true yarn start:mfe",
```
Comment (hubertzub-db) [inline mlflow/server/js/package.json:135]: so these commands are non-OSS (also one is FS-specific), however it doesn't seem to impose a big harm - on the other hand, it won't be easy to remove it from here without some specific customly made copybara commands

File: mlflow/server/js/src/common/sdk/UnityCatalogService.js [hunk 3]
```diff
@@ -5,4 +5,16 @@ export class UnityCatalogService {
    * List all UC catalogs.
    */
   static listCatalogs = () => getJson({ relativeUrl: 'ajax-api/2.0/unity-catalog/catalogs' });
+
+  /**
```
Comment (hubertzub-db) [inline mlflow/server/js/src/common/sdk/UnityCatalogService.js:9]: shall we OSS-guard UC-specific code? 🤔

**Review by hubertzub-db — APPROVED** [review]
code seems to be mostly ok, left one remark and one question about yarn commands and UC catalog respectively

File: mlflow/server/js/package.json (hunk 2 above)
Comment (harupy) [reply mlflow/server/js/package.json:135]: Yeah, I think it's fine to keep these commands unless they are harmful or contain confidential information.

File: mlflow/server/js/src/common/sdk/UnityCatalogService.js (hunk 3 above)
Comment (harupy) [reply mlflow/server/js/src/common/sdk/UnityCatalogService.js:9]: Yep :) I'll exclude this file from the sync files.

**Pull request merged**